platform/nvidia-bluefield/platform-api/sonic_platform/utils.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import functools
+import subprocess
+from sonic_py_common.logger import Logger
+
+logger = Logger()
+
+
+def read_only_cache():
```

Comment:
FYI, Python 3.9 and newer have this functionality in the stdlib: https://docs.python.org/3/library/functools.html#functools.cache